### ISSUE TITLE
Check if the replay exists before trying to show the replay preview. If there's replay put a nice message like 'the replay might've been deleted'

### ISSUE THREAD

**Issue by ryan953** [getsentry/sentry#43965]

**Comment by ryan953** [comment]
This is less important now that `useReplayData` is updated: https://github.com/getsentry/sentry/pull/44561

Prior to that change we would request the replay record, and also send off 4 requests for segment data, unconditionally, in parallel. This created a bunch of network traffic.

Now we're fetching the replay record first, and then asking for only as much segment data as exists. No more speculative requests.

The using messaging is still nice to have.
